leveldb.c

```diff
@@ -1199,10 +1208,11 @@ static void leveldb_iterator_dtor(zend_object_iterator *iter)
 	zval *object = &iterator->intern.data;
 	zval_ptr_dtor(object);
 
-	if (iterator->current) {
-		zval_ptr_dtor(iterator->current);
-		efree(iterator->current);
-	}
+		if (iterator->current) {
+			zval_ptr_dtor(iterator->current);
+			efree(iterator->current);
+			iterator->current = NULL;
```

Comment:
Not convinced this is actually necessary, the dtor shouldn't be called multiple times